.github/workflows/create-release-pr.yml

```diff
@@ -53,11 +53,38 @@ jobs:
         uses: actions/github-script@f28e40c7f34bde8b3046d885e986cb6290c5673b # v7.1.0
         with:
           script: |
-            const { data } = await github.rest.repos.generateReleaseNotes({
+            // --- Get latest release (previous tag) ---
+            let lastTag = "";
+            try {
+              const { data: latestRelease } = await github.rest.repos.getLatestRelease({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+              });
+              lastTag = latestRelease.tag_name;
+              core.info(`Previous release found: ${lastTag}`);
+            } catch (error) {
+              core.info("No previous release found — this will be the first release.");
+            }
+
+            // --- Get default branch ---
+            const { data: repoData } = await github.rest.repos.get({
               owner: context.repo.owner,
               repo: context.repo.repo,
-              tag_name: `v${process.env.PACKAGE_VERSION}`
             });
+            const defaultBranch = repoData.default_branch;
+            core.info(`Default branch: ${defaultBranch}`);
+
+            // --- Generate release notes ---
+            const params = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: `v${process.env.PACKAGE_VERSION}`,
+              target_commitish: defaultBranch,
+            };
+            if (lastTag) {
+              params.previous_tag_name = lastTag;
+            }
+            const { data } = await github.rest.repos.generateReleaseNotes(params);
```

Comment:
I would like to avoid writing JavaScript directly into the workflow file as much as possible. If possible, could we prepare this part as a script and have it called from here? I believe there would be the following advantages:
- Syntax highlighting in the editor would allow us to catch small bugs more quickly.
- Concerns can be separated between the script's standalone operation and the workflow, making verification and testing easier.

What do you think? I'd like to hear your opinions.